### PR TITLE
core/priority: refactor priority protocol

### PR DESCRIPTION
**Pull request by corverroos**
Refactors the priority protocol:
- Removes metadata from proto messages since not required anymore
- Makes all types generic `anypb.Any` to support arbitrary use-cases.
- Refactor static msgProvider to provide when triggered.

category: refactor
ticket: #1257

### CODE REVIEW

File: core/priority/calculate.go [hunk 1]
```diff
@@ -118,29 +166,41 @@ func validateMsgs(msgs []*pbv1.PriorityMsg) error {
...
+		} else if !proto.Equal(inst, instance) {
+			return errors.New("mismatching instances")
 		}
 		if dedupPeers(msg.PeerId) {
 			return errors.New("duplicate peer")
 		}
-		dedupTopics := newDeduper[string]() // Peers may not provide duplicate topics.
+		dedupTopics := newDeduper[[32]byte]() // Peers may not provide duplicate topics.
 		for _, topic := range msg.Topics {
-			if dedupTopics(msg.PeerId) {
```
Comment (corverroos) [inline core/priority/calculate.go:135]: bug!

File: core/priority/calculate.go [hunk 2]
```diff
@@ -37,28 +41,40 @@ func calculateResult(msgs []*pbv1.PriorityMsg, minRequired int) (*pbv1.PriorityR
...
 	// Calculate cluster wide resulting priorities by topic.
 	var topicResults []*pbv1.PriorityTopicResult
-	for topic, prioritySet := range prioritySetsByTopic {
+	for _, proposals := range proposalsByTopic {
 		// Calculate overall score for all priorities in the topic
 		// which effectively orders by count then by overall priority.
 		var (
-			scores        = make(map[string]int)
-			allPriorities []string
+			scores        = make(map[[32]byte]int)
```
Comment (xenowits) [inline core/priority/calculate.go:62]: can we give `[32]byte` a type? will make it more readable

File: core/priority/prioritiser.go [hunk 3]
```diff
@@ -33,13 +34,29 @@ import (
...
+// Instance identifies an instance of the priority protocol.
+type Instance proto.Message
+
+// Topic groups priorities in an instance.
+type Topic proto.Message
+
+// Priority is one of many grouped by a Topic being prioritised in an Instance.
+type Priority proto.Message
+
+// instanceKey is the hash of an Instance proto.Message and is used to index instanceData. See hashProto.
```
Comment (xenowits) [inline core/priority/prioritiser.go:46]:
```suggestion
// instanceKey is the hash of an Instance and is used to index instanceData. See hashProto.
```

Comment (corverroos) [reply core/priority/prioritiser.go:46]: think the original is fine

File: core/priority/calculate.go (hunk 2 above)
Comment (corverroos) [reply core/priority/calculate.go:62]: yeah not sure, I actually removed the `instanceKey` type, reverting to just `[32]byte`. I think defining types is mostly useful for the scope of the type is very large. In this case, it is pretty small.

**Pull request merged**